Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/SortAdapter.cpp

```diff
@@ -38,6 +39,60 @@ namespace AZ::DocumentPropertyEditor
         }
     }
 
+    Dom::Path RowSortAdapter::MapPath(const Dom::Path& path, bool mapToSource)
+    {
+        if (!m_sortActive)
+        {
+            return path;
+        }
+
+        Dom::Path mappedPath;
+        const auto* currNode = m_rootNode.get();
+
+        // go through each entry in the path, looking for an existing mapping at the give index
+        for (const auto& pathEntry : path)
+        {
+            if (pathEntry.IsIndex())
+            {
+                bool found = false;
+                for (auto [indexSortedNode, adapterSortedNode] : AZStd::views::zip(currNode->m_indexSortedChildren, currNode->m_adapterSortedChildren))
+                {
+                    const SortInfoNode* comparisonNode = (mapToSource ? adapterSortedNode : indexSortedNode.get());
+                    if (comparisonNode->m_domIndex == pathEntry.GetIndex())
+                    {
+                        // set the mapped entry
+                        const auto mappedIndex = (mapToSource ? indexSortedNode->m_domIndex : adapterSortedNode->m_domIndex);
+                        mappedPath.Push(mappedIndex);
+                        currNode = comparisonNode;
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found)
+                {
+                    // if there's no mapping, this *should* be a column entry. Pass it through as-is
+                    mappedPath.Push(pathEntry);
+                }
+            }
+            else
+            {
+                // RowSortAdapter only affect index entries, pass other types (like attribute strings) through
+                mappedPath.Push(pathEntry);
+            }
+        }
+        return mappedPath;
+    }
+
+    Dom::Path RowSortAdapter::MapFromSourcePath(const Dom::Path& sourcePath)
```

Comment:
question: Could this and the function it calls be `const`? It doesn't _look_ like any member data is mutated.